File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int solve(int i, int n, int t[]){
        if(i==n) return 1;
        if(i>n) return 0;
        if(t[i]!=0) return t[i];
        t[i] = solve(i+1,n,t) + solve(i+2, n,t);
        return t[i];
    }
    int climbStairs(int n) {
        int t[n+1];
        memset(t,0,sizeof(t));
        return solve(0, n, t);
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    int solve(int []dp, int n, int i){
        if(i==n) return 1;
        if(i>n) return 0;
        if(dp[i]!=0) return dp[i];
        return dp[i] = solve(dp, n, i+1) + solve(dp, n, i+2);
    }
    public int climbStairs(int n) {

        int dp[] = new int[n+1];
        
        return solve(dp, n, 0);
    }
}